numpy/tests/test_lazyloading.py

```diff
@@ -1,38 +1,41 @@
+import subprocess
 import sys
-from importlib.util import LazyLoader, find_spec, module_from_spec
 
 import pytest
 
+from numpy.testing import IS_WASM
+
 
-# Warning raised by _reload_guard() in numpy/__init__.py
-@pytest.mark.filterwarnings("ignore:The NumPy module was reloaded")
+@pytest.mark.skipif(IS_WASM, reason="can't start subprocess")
 def test_lazy_load():
     # gh-22045. lazyload doesn't import submodule names into the namespace
-    # muck with sys.modules to test the importing system
-    old_numpy = sys.modules.pop("numpy")
-
-    numpy_modules = {}
-    for mod_name, mod in list(sys.modules.items()):
-        if mod_name[:6] == "numpy.":
-            numpy_modules[mod_name] = mod
-            sys.modules.pop(mod_name)
-
-    try:
-        # create lazy load of numpy as np
-        spec = find_spec("numpy")
-        module = module_from_spec(spec)
-        sys.modules["numpy"] = module
-        loader = LazyLoader(spec.loader)
-        loader.exec_module(module)
-        np = module
-
-        # test a subpackage import
-        from numpy.lib import recfunctions  # noqa: F401
-
-        # test triggering the import of the package
-        np.ndarray
-
-    finally:
-        if old_numpy:
-            sys.modules["numpy"] = old_numpy
-            sys.modules.update(numpy_modules)
+
+    # Test within a new process, to ensure that we do not mess with the
+    # global state during the test run (could lead to cryptic test failures).
+    # This is generally unsafe, especially, since we also reload the C-modules.
+    code = r"""\
+import sys
+from importlib.util import LazyLoader, find_spec, module_from_spec
+
+# create lazy load of numpy as np
+spec = find_spec("numpy")
+module = module_from_spec(spec)
+sys.modules["numpy"] = module
+loader = LazyLoader(spec.loader)
+loader.exec_module(module)
+np = module
+
+# test a subpackage import
+from numpy.lib import recfunctions  # noqa: F401
+
+# test triggering the import of the package
+np.ndarray
+"""
+    p = subprocess.run(
+        (sys.executable, '-c', code),
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        encoding='utf-8',
+        check=False,
+    )
+    assert p.returncode == 0, p.stdout
```

Comment:
@dschult if you can have a quick look: I think doing the lazy loading test in a subprocess is identical (but safer as it avoids messing with `sys.modules`).  Does that seem right?